topiary-cli/src/main.rs

```diff
@@ -158,16 +197,35 @@ async fn run() -> CLIResult<()> {
         }
     };
 
-    formatter(
-        &mut input,
-        &mut output,
-        &query,
-        language,
-        &grammar,
-        operation,
-    )?;
-
-    output.into_inner()?.persist()?;
+    let tasks: Vec<_> = fmt_args
+        .into_iter()
+        .map(|(input, output, language, grammar, query)| {
+            tokio::spawn(async move {
+                let mut input: Box<dyn Read> = match input.as_str() {
+                    "-" => Box::new(stdin()),
+                    file => Box::new(BufReader::new(File::open(file).unwrap())),
+                };
+                let mut output: BufWriter<OutputFile> =
+                    BufWriter::new(OutputFile::new(&output).unwrap());
+
+                formatter(
+                    &mut input,
+                    &mut output,
+                    &query,
+                    &language,
+                    &grammar,
+                    operation,
+                )
+                .unwrap();
+
+                output.into_inner().unwrap().persist().unwrap();
+            })
+        })
+        .collect();
+
+    for task in tasks {
+        task.await.unwrap();
```

Comment:
Isn't this just running them sequentially? Perhaps use `par_iter()` from Rayon?

Reply:
I don't think so, this is from the documentation:
> The provided future will start running in the background immediately when spawn is called, even if you don’t await the returned JoinHandle.

https://docs.rs/tokio/latest/tokio/task/fn.spawn.html